### PR TITLE
docs: applications and authorization

### PR DESCRIPTION
**Pull request by mtojek**
Related: https://github.com/coder/coder/issues/3522

Changes:
* Documented: `Applications`, `Authorization`
* Fix: missing description for structures (fixed with `@Description`)
* Fix: replace `@Consume` with `@Accept`

Question:
~~* What quality level of documentation do we want to achieve? (description: `none` vs. `DisplayName is the displayed name`)~~ Fix: `s/none//g`

### CODE REVIEW

**Review by mafredri — COMMENTED** [review]
> * What quality level of documentation do we want to achieve? (description: `none` vs. `DisplayName is the displayed name`)

Instead of `none`, we could omit the text entirely (empty). I think it's fine not to document all obvious things for now. But ultimately I'd like for us to have very good coverage in `codersdk`/API docs with explanations for most fields.

File: coderd/authorize.go [hunk 1]
```diff
@@ -104,6 +104,16 @@ func (h *HTTPAuthorizer) AuthorizeSQLFilter(r *http.Request, action rbac.Action,
...
 
+// @Summary Check authorization
+// @ID check-authorization
+// @Security CoderSessionToken
+// @Accept json
+// @Produce json
+// @Tags Authorization
+// @Param request body codersdk.AuthorizationRequest true "Authorization request"
+// @Success 200 {object} codersdk.AuthorizationResponse
+// @Router /authcheck [post]
```
[the remark this answers is not in the document]
Comment (mtojek) [reply coderd/authorize.go:118]: Fixed

File: codersdk/parameters.go [hunk 2]
```diff
@@ -83,7 +83,7 @@ type ParameterSchema struct {
 	ValidationContains []string `json:"validation_contains,omitempty"`
 }
 
-// CreateParameterRequest is used to create a new parameter value for a scope.
+// @Description CreateParameterRequest is a structure used to create a new parameter value for a scope.
```
[the remark this answers is not in the document]
Comment (mtojek) [reply codersdk/parameters.go:88]: Fixed

File: docs/api/schemas.md [hunk 3]
```diff
@@ -2,6 +2,109 @@
...
+| ------------------ | ---------------------------------------------------------- | -------- | ------------ | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `checks`           | object                                                     | false    | none         | Checks is a map keyed with an arbitrary string to a permission check.<br>The key can be any string that is helpful to the caller, and allows<br>multiple permission checks to be run in a single request.<br>The key ensures that each permission check has the same key in the<br>response. |
+| » `[any property]` | [codersdk.AuthorizationCheck](#codersdkauthorizationcheck) | false    | none         | » **additionalproperties** is used to check if the currently authenticated user (or the specified user) can do a given action to a given set of objects.                                                                                                                                     |
```
[the remark this answers is not in the document]
Comment (mtojek) [reply docs/api/schemas.md]: Fixed, but I just used `It`. This way we don't need to deal with singular/plural form: `The properties is used...`

File: docs/api/applications.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+## Redirect to URI with encrypted API key
+
+### Code samples
+
+```shell
+# Example request using curl
+curl -X GET http://coder-server:8080/api/v2/applications/auth-redirect \
+  -H 'Coder-Session-Token: API_KEY'
+
```
[the remark this answers is not in the document]
Comment (mtojek) [reply docs/api/applications.md]: Fixed

File: docs/api/schemas.md [hunk 5]
```diff
@@ -2,6 +2,109 @@
...
+| `owner_id`        | string | false    | none         | Owner id (optional) adds the set constraint to all resources owned by a given user.                                                                                                                                                                                                                                                                                                |
+| `resource_id`     | string | false    | none         | Resource id (optional) reduces the set to a singular resource. This assigns<br>a resource ID to the resource type, eg: a single workspace.<br>The rbac library will not fetch the resource from the database, so if you<br>are using this option, you should also set the `OwnerID` and `OrganizationID`<br>if possible. Be as specific as possible using all the fields relevant. |
```
[the remark this answers is not in the document]
Comment (mtojek) [reply docs/api/schemas.md]: Fixed

**Comment by mtojek** [comment]
Thanks for the review, @mafredri! All comments addressed. It's interesting that we can continuously improve the rendering procedure to achieve better results!

**Review by mafredri — APPROVED** [review]
Nice work!

> It's interesting that we can continuously improve the rendering procedure to achieve better results!

I agree! It's nice to have bite-sized pieces, easier to catch oddities. 😄

**Pull request merged**